vortex-datafusion/src/persistent/opener.rs

```diff
@@ -105,10 +105,23 @@ impl FileOpener for VortexFileOpener {
                 // DataFusion likes ~8k row batches. Ideally we would respect the config,
                 // but at the moment our scanner has too much overhead to process small
                 // batches efficiently.
-                .with_split_by(SplitBy::RowCount(8 * batch_size))
+                .with_split_by(SplitBy::Layout)
                 .map_to_record_batch(projected_arrow_schema.clone())
                 .into_stream()?
+                .map_ok(move |rb| {
+                    let mut results = vec![];
```

Comment:
Remove the vec, return an iter/stream

Reply:
making this change becomes extremely verbose and I can't figure out anything that actually compiles. I can spend some more time on it but it doesn't seem that important? the iter/stream we allocate here will be the same size.

Reply:
as I was writing that I figured out how to do that 🤦